Plugins/GitUIPluginInterfaces/ManagedExtensibility.cs

```diff
@@ -1,97 +1,91 @@
 ﻿using System;
 using System.Collections.Generic;
-using System.ComponentModel.Composition.Hosting;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
 using System.Reflection;
+using System.Threading;
 using System.Windows.Forms;
+using GitUI;
+using Microsoft.VisualStudio.Composition;
 
 namespace GitUIPluginInterfaces
 {
     public static class ManagedExtensibility
     {
-        private static List<CompositionContainer> _compositionContainers;
+        private static readonly Lazy<ExportProvider> _exportProvider = new Lazy<ExportProvider>(CreateExportProvider, LazyThreadSafetyMode.ExecutionAndPublication);
 
-        private static readonly object compositionContainerSyncObj = new object();
+        private static ExportProvider CreateExportProvider()
+        {
+            var stopwatch = Stopwatch.StartNew();
 
-        private static readonly HashSet<string> _loggedExceptionMessages = new HashSet<string>();
```

Comment:
:memo: This logging was added after I originally implemented the vs-mef change. I removed it during the current rebase, but it could be added back if necessary.